src/components/admin/Billing.tsx

```diff
@@ -0,0 +1,460 @@
+import {
+    Box,
+    Button,
+    CircularProgress,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    Divider,
+    FormControl,
+    InputLabel,
+    MenuItem,
+    Select,
+    Skeleton,
+    Stack,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    Typography,
+} from '@mui/material';
+import {
+    AddressElement,
+    Elements,
+    PaymentElement,
+    useElements,
+    useStripe,
+} from '@stripe/react-stripe-js';
+import { loadStripe } from '@stripe/stripe-js';
+import { Auth } from '@supabase/ui';
+import {
+    deleteTenantPaymentMethod,
+    getSetupIntentSecret,
+    getTenantPaymentMethods,
+    setTenantPrimaryPaymentMethod,
+} from 'api/billing';
+import { authenticatedRoutes } from 'app/routes';
+import AdminTabs from 'components/admin/Tabs';
+import MessageWithLink from 'components/content/MessageWithLink';
+import useCombinedGrantsExt from 'hooks/useCombinedGrantsExt';
+import usePageTitle from 'hooks/usePageTitle';
+import useTenants from 'hooks/useTenants';
+import { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { getUserDetails } from 'services/supabase';
+import AmexLogo from '../../images/payment-methods/amex.png';
+import DiscoverLogo from '../../images/payment-methods/discover.png';
+import MastercardLogo from '../../images/payment-methods/mastercard.png';
+import VisaLogo from '../../images/payment-methods/visa.png';
```

Comment:
Did these come from Stripe?

Reply:
Stripe no longer directly distributes icons, instead they [refer you to someone else](https://support.stripe.com/questions/where-to-find-logos-for-accepted-credit-card-types). I downloaded the icons from here: http://www.dreamstale.com/free-download-payment-method-vector-icons/